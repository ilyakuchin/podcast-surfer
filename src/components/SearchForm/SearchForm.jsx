import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import history from '../../helpers/history';
import { setSearchPhrase } from '../../redux/actions/SearchPhrase/searchPhrase';
import { searchPodcasts } from '../../redux/actions/Podcasts/podcasts';

export function SearchForm({
  searchPhrase,
  jwt,
  setSearchPhraseConnect,
  searchPodcastsConnect,
}) {
  return (
    <form
      onSubmit={(e) => {
        searchPodcastsConnect(e, searchPhrase, jwt);
        history.push('/search');
      }}
    >
      <input
        value={searchPhrase}
        onChange={setSearchPhraseConnect}
        action={{
          icon: 'search',
        }}
        placeholder='Search...'
      />
    </form>
  );
}

const mapStateToProps = (state) => {
  return {
    searchPhrase: state.searchPhrase,
    jwt: state.userInfo.jwt,
  };
};

const dispatchStateToProps = (dispatch) => {
  return {
    searchPodcastsConnect: (e, searchPhrase, jwt) =>
      dispatch(searchPodcasts(e, searchPhrase, jwt)),
    setSearchPhraseConnect: (e) => dispatch(setSearchPhrase(e.target.value)),
  };
};

SearchForm.propTypes = {
  searchPhrase: PropTypes.string.isRequired,
  jwt: PropTypes.string.isRequired,
  setSearchPhraseConnect: PropTypes.func.isRequired,
  searchPodcastsConnect: PropTypes.func.isRequired,
};

const ConnectedSearchForm = connect(
  mapStateToProps,
  dispatchStateToProps
)(SearchForm);

export default ConnectedSearchForm;
