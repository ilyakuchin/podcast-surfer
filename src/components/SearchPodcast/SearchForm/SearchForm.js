import React, { Component } from "react";
import styled from "styled-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import axios from "axios";

const Search = styled.form`
  margin-top: 30px;
  grid-area: search;
  display: grid;
  max-width: 700px;
  align-self: center;
  justify-self: center;
  grid-template-columns: 10fr 1fr;
  grid-template-areas: "input button";
`;

const SearchInput = styled.input`
  grid-area: input;
  font-size: 18px;
`;

const SearchButton = styled.button`
  grid-area: button;
  font-size: 18px;
`;

export default class SearchForm extends Component {
  render() {
    return (
      <Search>
        <SearchInput
          type="text"
          placeholder="Search podcast.."
          name="search"
          onChange={this.props.handleChange}
        />
        <SearchButton
          type="submit"
          onClick={e => {
            e.preventDefault();
            this.props.clearPodcastResults();
            axios
              .get(
                `https://podcast-player-api.herokuapp.com/podcasts?name=${this.props.searchPhrase}`
              )
              .then(res => {
                this.props.updatePodcastResults(res.data);
              });
          }}
        >
          <FontAwesomeIcon icon={faSearch} />
        </SearchButton>
      </Search>
    );
  }
}
