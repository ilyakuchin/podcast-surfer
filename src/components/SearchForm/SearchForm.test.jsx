import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SearchForm } from './SearchForm';

configure({ adapter: new Adapter() });

describe('Test SearchForm component', () => {
  test('should render correctly', () => {
    expect(
      toJson(
        shallow(
          <SearchForm
            searchPhrase='searchPhrase'
            jwt='test'
            setSearchPhrase={null}
            searchPodcasts={null}
          />
        )
      )
    ).toMatchSnapshot();
  });
});
