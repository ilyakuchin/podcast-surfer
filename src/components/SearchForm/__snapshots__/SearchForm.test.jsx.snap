// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchForm component should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    action={
      Object {
        "icon": "search",
      }
    }
    onChange={[Function]}
    placeholder="Search..."
    value="searchPhrase"
  />
</form>
`;
